mocca-jaxrs2/src/main/java/com/paypal/mocca/client/MoccaJaxrsClient.java

```diff
@@ -47,97 +57,113 @@ public Client build() {
 
         @Override
         public ClientBuilder withConfig(Configuration config) {
-            throw new UnsupportedOperationException(USAGE_ERR_MSG);
+            log.warn(USAGE_ERR_MSG);
+            return this;
         }
 
         @Override
         public ClientBuilder sslContext(SSLContext sslContext) {
-            throw new UnsupportedOperationException(USAGE_ERR_MSG);
+            log.warn(USAGE_ERR_MSG);
+            return this;
         }
 
         @Override
         public ClientBuilder keyStore(KeyStore keyStore, char[] password) {
-            throw new UnsupportedOperationException(USAGE_ERR_MSG);
+            log.warn(USAGE_ERR_MSG);
+            return this;
         }
 
         @Override
         public ClientBuilder trustStore(KeyStore trustStore) {
-            throw new UnsupportedOperationException(USAGE_ERR_MSG);
+            log.warn(USAGE_ERR_MSG);
+            return this;
         }
 
         @Override
         public ClientBuilder hostnameVerifier(HostnameVerifier verifier) {
-            throw new UnsupportedOperationException(USAGE_ERR_MSG);
+            log.warn(USAGE_ERR_MSG);
+            return this;
         }
 
         @Override
         public ClientBuilder executorService(ExecutorService executorService) {
-            throw new UnsupportedOperationException(USAGE_ERR_MSG);
+            log.warn(USAGE_ERR_MSG);
+            return this;
         }
 
         @Override
         public ClientBuilder scheduledExecutorService(ScheduledExecutorService scheduledExecutorService) {
-            throw new UnsupportedOperationException(USAGE_ERR_MSG);
+            log.warn(USAGE_ERR_MSG);
+            return this;
         }
 
         @Override
         public ClientBuilder connectTimeout(long timeout, TimeUnit unit) {
-            throw new UnsupportedOperationException(USAGE_ERR_MSG);
+            return this;
         }
 
         @Override
         public ClientBuilder readTimeout(long timeout, TimeUnit unit) {
-            throw new UnsupportedOperationException(USAGE_ERR_MSG);
+            return this;
```

Comment:
Can we add a log warning message here before returning?

Reply:
The reason I removed this is because, given feign's current impl, it will always be called.  I hate to log a warning message when I coded it to happen.

Do you still think that?  Logging at trace/debug are other options.

Reply:
I see. I agree with you, a warning here would be terrible.
What if if we do this:

``` java
if (log.isDebugEnabled()) {
// log a warning
}
```

Reply:
I see. Agree with you, a warning here would be terrible then. I think debug would better then.

Reply:
I _almost_ pushed that, but I didn't given what I logged in the constructor, which honestly I think should be moved to `info`.

Reply:
I hope I didn't wear you down:)  I meant I already had that code written in my last commit, but ultimately removed it.  You making this comment had me thinking again.  I just wasn't sure you had seen the constructor log message.

Anyhow, I'm going to change that constructor message to `info` in a another PR.  If you think we should have debug messages here, you just need to let me know.  I'm like 49.999% no and 50.001% yes!

Reply:
hahah no worries. Actually I got confused. I thought your last comment meant "PR is ready to be merged", but now I see it actually meant "I need to change log level in the constructor". Sorry about that :-)

So, that is the bug you were referring to? Or is there anything else besides that?